Add front-end design for project

etcadinfinitum opened the pull request:
I'm really owning the retro computing aesthetic theme of this event, no regrets :grin:

etcadinfinitum committed:
Add page skeleton :skull:

etcadinfinitum committed:
Add page banner and menu divs :bird:

This website will be my magnum opus, my masterpiece, my muse.

Also, the menu is responsive and will collapse on mobile screens.

etcadinfinitum committed:
Add basic .gitignore for Vim tmp files :mushroom:

etcadinfinitum committed:
Add function to toggle visible divs :fire:

etcadinfinitum committed:
Change menu item bg color when selected :rainbow:

Trying to use as many of the named CSS colors as possible >:]

bpas247 commented on website/main.js:
```diff
@@ -0,0 +1,20 @@
+function renderPage(divName, menuItem) {
+    // hide any visible content divs
+    var contentDivs = document.getElementsByClassName('content');
```
I would recommend never using `var`. You should use `let` or `const` instead.

bpas247 commented on website/main.js:
```diff
@@ -0,0 +1,20 @@
+        if (!contentDivs[i].classList.contains('hidden')) {
+            contentDivs[i].classList.add('hidden');
+        }
+    }
```
I'm personally a fan of writing declarative JS, so I would use `forEach` here.
```suggestion
    Array.from(contentDivs).forEach(contentDiv => {
        if (!contentDiv.classList.contains('hidden')) {
            contentDiv.classList.add('hidden');
        }
    });
```

bpas247 commented on website/main.js:
```diff
@@ -0,0 +1,20 @@
+    var menuItems = document.getElementsByClassName('menu-item');
+    for (var i = 0; i < menuItems.length; i++) {
+        menuItems[i].classList.remove('active-task');
+    }
```
Same as above comment.
```suggestion
    Array.from(menuItems).forEach(menuItem => {
        menuItem.classList.remove('active-task');
    });
```

bpas247 reviewed (commented):
Just a few minor suggestions. Since you already have JQuery as a dependency, I would recommend using that than using vanilla JS 👍

bpas247 commented on website/index.html:
```diff
@@ -0,0 +1,41 @@
+    </div>
+
+    <div id="teachUs" class="content hidden">
+    </div>
```
Last side note: it's not really best practice to hide DOM elements. If you are wanting to hide them, they should just be removed from the DOM completely. For this use case, we could just use a single DOM element and mutate that whenever the end user click on one of the three menu item elements

etcadinfinitum replied:
@bpas247 can you link to code examples or tutorials for this? This is bleeding into the realm of stuff that I don't know how to do, so some examples would be helpful.

bpas247 replied:
Eh, on second thought this is probably fine. Removing DOM nodes would mess up the layout, and make styling more difficult

etcadinfinitum committed:
Switched to declarative forEach loop

Co-Authored-By: Brady Pascoe <18705892+bpas247@users.noreply.github.com>

etcadinfinitum committed:
Switched to declarative forEach loop

Co-Authored-By: Brady Pascoe <18705892+bpas247@users.noreply.github.com>

etcadinfinitum committed:
Add AJAX calls to retrieve all crow species data

Using the `/test` stage for the API gateway resources; however, the
stuff works!!

etcadinfinitum committed:
Add dynamic content generation for crow species pg

etcadinfinitum committed:
Add Reset button to clear local storage of data

etcadinfinitum committed:
Give the crobbos some style :nail_care:

etcadinfinitum committed:
Move JavaScript files into subdirectory

etcadinfinitum committed:
Add basic 404 page :hankey:

etcadinfinitum committed:
Add functionality for submitting facts :squirrel:

etcadinfinitum committed:
Add fact retrieval for user facts :toilet:

etcadinfinitum commented:
My body is ready for this PR review :toilet: :hankey:

etcadinfinitum committed:
Fix small things (nothing to see here :eyes:)

etcadinfinitum committed:
Improve layout of retrieved facts cards :boom:

etcadinfinitum committed:
Add sweet title to banner div

:zOOM:

Fixes #10

etcadinfinitum committed:
Fix user fact submission page :question:

Mostly worked to prevent duplicate entries of the submission status
from multiple submissions.

Also did some style stuff to make the form a bit easier to use.

Fixes #7

etcadinfinitum committed:
Parse the stringified form input for displaying

Fixes #8

Chris-Johnston commented on website/scripts/populate.js:
```diff
@@ -0,0 +1,99 @@
+        str += "\nYour input overwrote the input previously added by '" + response['body']['Attributes']['source'] + "', another crow expert.";
+    }
+    let msgDiv = document.getElementById("factRequestPlaceholder");
+    msgDiv.innerHTML = str;
```
should also sanitize here: https://owasp.org/www-community/attacks/xss/

etcadinfinitum replied:
There is an open issue for this: #5 

Definitely on the short list of things to do pronto

Chris-Johnston commented on website/styles.css:
```diff
@@ -0,0 +1,131 @@
+}
+
+#teachUs {
+    background: lightcyan;
```
```suggestion
    background: azure;
```
`0xE0FFFF` vs `0xF0FFFF`, similar enough

etcadinfinitum replied:
lol, aight

Chris-Johnston replied:
hehe

Chris-Johnston reviewed (commented):
The `pretty.png` file doesn't seem to be used. I'm a bit iffy on adding copyrighted stuff (that isn't ours) to a repo that's MIT licensed.

Chris-Johnston commented on website/404.html:
```diff
@@ -0,0 +1,18 @@
+    <meta charset="utf-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
+    <link rel="stylesheet" href="styles.css"/>
+    <title>Crow Fact Oracle&trade;</title>
```
```suggestion
    <title>Crow Fact Oracle</title>
```
Not sure what the `&trade;` is?

etcadinfinitum replied:
It's the trademark symbol! :D
![tm](https://user-images.githubusercontent.com/29528527/78540134-8783e300-77a8-11ea-867d-7b3e35ab3a84.png)

etcadinfinitum committed:
Update website/styles.css

Co-Authored-By: Chris Johnston <16418643+Chris-Johnston@users.noreply.github.com>

etcadinfinitum committed:
Remove unused (copyrighted) image

...whoops
